### ISSUE TITLE
BibTeX parser hangs

### ISSUE THREAD

**Issue by jayvdb** [brechtm/citeproc-py#53]
After converting the [this MODS](https://e-publications.une.edu.au/fedora/get/une:13794/MODS) to bib using `bibutils` tool `mods2bib` v5.6, the output is

``` bibtex

@Misc{pes:5137,
author="Cao, Daomin and Noussair, E. S. and Yan, Shusen",
title="Existence and Nonexistence of Interior-Peaked Solution for a Nonlinear Neumann Problem",
year="2001",
publisher="University of California",
address="Berkeley, United States of America",
volume="200",
number="1",
pages="19--42",
keywords="Ordinary Differential Equations, Difference Equations and Dynamical Systems",
abstract="We show that the critical problem \{-∆$\upsilon$ + $\lambda$$\upsilon$ = $\upsilon${\texttwosuperior}*⁻{\textonesuperior} + $\alpha$$\upsilon$q⁻{\textonesuperior}, $\upsilon$ > 0 in $\Omega$, ∂$\upsilon$∕∂$\nu$ = 0 on ∂$\Omega$, 2 < q < 2* = 2N/(N − 2), has no positive solutions concentrating, as $\lambda$ {\textrightarrow} ∞, at interior points of $\Omega$ if a = 0, but for a class of symmetric domains $\Omega$, the problem with $\alpha$ > 0 has solutions concentrating at interior points of $\Omega$.",
note="C1",
note="Yes",
issn="0030-8730",
doi="10.2140/pjm.2001.200.19",
language="en"
}
```

This causes the parser to hang:

```
  File ".../citeproc/source/bibtex/bibtex.py", line 70, in __init__
    bibtex_database = BibTeXParser(filename, encoding)
  File ".../citeproc/source/bibtex/bibparse.py", line 48, in __init__
    self._parse(self.file)
  File ".../citeproc/source/bibtex/bibparse.py", line 54, in _parse
    entry = self._parse_entry(file)
  File ".../citeproc/source/bibtex/bibparse.py", line 102, in _parse_entry
    value = self._parse_value(file)
  File ".../citeproc/source/bibtex/bibparse.py", line 138, in _parse_value
    value = self._parse_string(file, char)
  File ".../citeproc/source/bibtex/bibparse.py", line 157, in _parse_string
    char = file.read(1)
  File "/usr/lib64/python2.7/codecs.py", line 299, in decode
    def decode(self, input, final=False):
KeyboardInterrupt
```

**Comment by jayvdb** [comment]
The problem is the abstract contains `..\{-...` , which comes from https://github.com/jayvdb/bibutils-archive/blame/master/lib/latex.c#L27 .

`bibparse.py` recognises that as increasing the depth of braces, and fails to find the end.

The real problem here is that it doesnt check whether the end of file was reached, which #59 addresses.

Then we can also use a one character look behind buffer to not increase the depth when encountering the escaped `\{` .

**Comment by yarikoptic** [comment]
#59 now was merged and present in 0.5.0 release, so I will assume that the core issue was resolved.  If not - please reopen
